zipline/data/history_loader.py

```diff
@@ -343,6 +358,20 @@ def _calendar(self):
     def _array(self, start, end, assets, field):
         pass
 
+    def _decimal_places_for_asset(self, asset, reference_date):
+        if isinstance(asset, Future) and asset.tick_size:
+            return number_of_decimal_places(asset.tick_size)
+        elif isinstance(asset, ContinuousFuture):
+            rf = self._roll_finders[asset.roll_style]
+            contract_sid = rf.get_contract_center(
+                asset.root_symbol, reference_date, asset.offset,
+            )
```

Comment:
Can you just use the asset finder for this?
```python
contract_sid = self._asset_finder.get_ordered_contracts(
    asset.root_symbol).contract_before_auto_close(reference_date.value)
```

Seems like the offset probably doesn't matter?